user-management/server/lib/authorization-middleware.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { NextFunction, Request, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import { ClientError } from './client-error.js';
+
+const hashKey = process.env.TOKEN_SECRET ?? '';
+if (!hashKey) throw new Error('TOKEN_SECRET not found in env');
+
+export function authMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+): void {
+  const auth = req.get('authorization');
+  const token = auth?.split('Bearer ')[1];
+  if (!auth || !token) throw new ClientError(401, 'authentication required');
```

Comment:
FYI, if auth is undefined then token will be also, so you only have to check one of the two. :-)